pallets/omnipool/src/tests/sell.rs

```diff
@@ -675,3 +676,209 @@ fn sell_lrna_should_fail_when_exceeds_max_out_ratio() {
+		.with_max_trade_volume_limit_per_block(TEN_PERCENT)
+		.build()
+		.execute_with(|| {
+			let min_limit = 10 * ONE;
+			let sell_amount = CircuitBreaker::calculate_limit(initial_liquidity, TEN_PERCENT).unwrap();
+
+			// Act & Assert
+			//Asset_out amount would be 1056_910_569_105_689 in a successful trade, but it fails due to limit
+			assert_noop!(
+				Omnipool::sell(Origin::signed(TRADER), DOT, ACA, sell_amount, min_limit),
+				pallet_circuit_breaker::Error::<Test>::MinTradeVolumePerBlockReached
+			);
+		});
+}
+
+#[test]
+fn sell_should_fail_when_consequent_trades_exceed_trade_volume_min_limit() {
+	// Arrange
+	let initial_liquidity = 10_000 * ONE;
+
+	ExtBuilder::default()
+		.with_endowed_accounts(vec![
+			(Omnipool::protocol_account(), DAI, 1000 * ONE),
+			(Omnipool::protocol_account(), HDX, NATIVE_AMOUNT),
+			(LP1, DOT, 2_000_000 * ONE),
+			(LP1, ACA, 2_000_000 * ONE),
+			(TRADER, DOT, 2_000_000 * ONE),
+		])
+		.with_registered_asset(DOT)
+		.with_registered_asset(ACA)
+		.with_initial_pool(FixedU128::from_float(0.5), FixedU128::from(1))
+		.with_token(DOT, FixedU128::from_float(0.65), LP1, initial_liquidity)
+		.with_token(ACA, FixedU128::from_float(0.50), LP1, initial_liquidity)
+		.with_max_trade_volume_limit_per_block(TEN_PERCENT)
+		.build()
+		.execute_with(|| {
+			let min_limit = 10 * ONE;
+			let sell_amount = CircuitBreaker::calculate_limit(initial_liquidity, FIVE_PERCENT).unwrap();
+
+			// Act & Assert
+			assert_ok!(Omnipool::sell(Origin::signed(TRADER), DOT, ACA, sell_amount, min_limit));
+
+			assert_noop!(
+				Omnipool::sell(Origin::signed(TRADER), DOT, ACA, sell_amount, min_limit),
+				pallet_circuit_breaker::Error::<Test>::MinTradeVolumePerBlockReached
+			);
+		});
+}
+
+#[test]
+fn trade_volume_limit_should_be_ignored_for_hub_asset() {
+	// Arrange
+	let initial_liquidity = 10_000 * ONE;
+	let aca_price = FixedU128::from_float(0.65);
+
+	ExtBuilder::default()
+		.with_endowed_accounts(vec![
+			(Omnipool::protocol_account(), DAI, 1_000 * ONE),
+			(Omnipool::protocol_account(), HDX, NATIVE_AMOUNT),
+			(LP1, DOT, 2_000_000 * ONE),
+			(LP1, ACA, 2_000_000 * ONE),
+			(TRADER, LRNA, 2_000_000 * ONE),
+		])
+		.with_registered_asset(DOT)
+		.with_registered_asset(ACA)
+		.with_initial_pool(FixedU128::from_float(0.5), FixedU128::from(1))
+		.with_token(DOT, FixedU128::from_float(0.65), LP1, initial_liquidity)
+		.with_token(ACA, aca_price, LP1, initial_liquidity)
+		.with_max_trade_volume_limit_per_block(TEN_PERCENT)
+		.build()
+		.execute_with(|| {
+			let min_limit = 10 * ONE;
+
+			let sell_amount =
+				CircuitBreaker::calculate_limit(aca_price.checked_mul_int(initial_liquidity).unwrap(), TEN_PERCENT)
+					.unwrap() + ONE;
+
+			// Act & Assert
+			assert_ok!(Omnipool::sell(
+				Origin::signed(TRADER),
+				LRNA,
+				ACA,
+				sell_amount,
+				min_limit
+			),);
+		});
+}
+
+//TODO Dani: add integration tests for complex cases
```

Comment:
Can be removed, some integration tests have been already added, and possible one more will be about liquidity addition